### ISSUE TITLE
Exported variables are not evaluated in some cases

### ISSUE THREAD

**Issue by AndreaOrru** [zig-lang/zig#378]
Exported variables must always be compiled because some external code (i.e., assembly) might depend on them.

However, this doesn't always happen. Consider the following case:

```
// syscall.zig
export var syscall_handlers = []fn() {
    ...
};
```

```
// kmain.zig
const syscall = @import("syscall.zig");

export fn kmain() {
    ...
    // Code where syscall is not referenced.
    ...
}
```

`syscall_handlers` wouldn't be compiled into the final executable.
However, this works:

```
// kmain.zig
use @import("syscall.zig");
...
```

**Comment by andrewrk** [comment]
You can also do `comptime { _= @import("foo.zig")}`

**Comment by andrewrk** [comment]
This is a design limitation that can't really be solved, unless you want to propose some fundamental change (which you are more than welcome to do).

But the fact is right now that we have full lazy evaluation of top level declarations, to the point where you can rely on this fact to avoid compile errors. For example:

```
const windows = @import("windows.zig");
fn foo() {
    if (builtin.os == builtin.Os.windows) {
        windows.bar();
    }
}
```

This code is supposed to build successfully on both Windows and non-Windows, and on non-Windows it should not import windows.zig.

So, long story short, if you have exported symbols, best bet is to import the file they're in explicitly with the `comptime` block shown in comment above from the root source file. This is equivalent to passing multiple .c or .o files to the C compiler, but at least here it's in the source code and not the command line.

**Comment by AndreaOrru** [comment]
Fair enough. :)
